File: src/main/java/com/esprit/models/Commande.java
package com.esprit.models;

import java.time.LocalDateTime;

public class Commande {
    private int id;
    private int clientId;
    private int equipementId;
    private String etat;
    private String statutPaiement;
    private int quantite;
    private LocalDateTime dateCreation;
    private Equipement equipement; // Ajout de l'attribut Equipement

    // Constructeurs
    public Commande(int id, int clientId, int equipementId, String etat, String statutPaiement, int quantite, LocalDateTime dateCreation) {
        this.id = id;
        this.clientId = clientId;
        this.equipementId = equipementId;
        this.etat = etat;
        this.statutPaiement = statutPaiement;
        this.quantite = quantite;
        this.dateCreation = dateCreation;
    }

    public Commande(int id, int clientId, int equipementId, String etat, String statutPaiement, int quantite) {
        this(id, clientId, equipementId, etat, statutPaiement, quantite, LocalDateTime.now());
    }

    public Commande(int clientId, int equipementId, String etat, String statutPaiement, int quantite) {
        this(0, clientId, equipementId, etat, statutPaiement, quantite, LocalDateTime.now());
    }

    // Getters et Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getClientId() {
        return clientId;
    }

    public void setClientId(int clientId) {
        this.clientId = clientId;
    }

    public int getEquipementId() {
        return equipementId;
    }

    public void setEquipementId(int equipementId) {
        this.equipementId = equipementId;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public String getStatutPaiement() {
        return statutPaiement;
    }

    public void setStatutPaiement(String statutPaiement) {
        this.statutPaiement = statutPaiement;
    }

    public int getQuantite() {
        return quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    public LocalDateTime getDateCreation() {
        return dateCreation;
    }

    public void setDateCreation(LocalDateTime dateCreation) {
        this.dateCreation = dateCreation;
    }

    public Equipement getEquipement() {
        return equipement;
    }

    public void setEquipement(Equipement equipement) {
        this.equipement = equipement;
    }

    @Override
    public String toString() {
        return "Commande{" +
                "id=" + id +
                ", clientId=" + clientId +
                ", equipementId=" + equipementId +
                ", etat='" + etat + '\'' +
                ", statutPaiement='" + statutPaiement + '\'' +
                ", quantite=" + quantite +
                ", dateCreation=" + (dateCreation != null ? dateCreation : "Non définie") +
                ", equipement=" + equipement +
                '}';
    }
}

File: src/Entity/Commande.php
<?php
// src/Entity/Commande.php

namespace App\Entity;

use App\Repository\CommandeRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: CommandeRepository::class)]
#[ORM\Table(name: 'commande')] 
class Commande
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: Client::class)]
    #[ORM\JoinColumn(name: 'clientId', referencedColumnName: 'id')]
    private ?Client $client = null;

    #[ORM\Column(name: 'etat', type: 'string', length: 50)]
    private ?string $etat = null;

    #[ORM\Column(name: 'statutPaiement', type: 'string', length: 50)]
    private ?string $statutPaiement = null;

    #[ORM\ManyToOne(targetEntity: Equipement::class)]
    #[ORM\JoinColumn(name: 'equipementId', referencedColumnName: 'id')]
    private ?Equipement $equipement = null;

    #[ORM\Column(name: 'quantite', type: 'integer')]
    private ?int $quantite = null;

    #[ORM\Column(name: 'dateCreation', type: 'datetime')]
    private ?\DateTimeInterface $dateCreation = null;

    // Getters et Setters

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getClient(): ?Client
    {
        return $this->client;
    }

    public function setClient(Client $client): self
    {
        $this->client =$client;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): self
    {
        $this->etat = $etat;

        return $this;
    }

    public function getStatutPaiement(): ?string
    {
        return $this->statutPaiement;
    }

    public function setStatutPaiement(string $statutPaiement): self
    {
        $this->statutPaiement = $statutPaiement;

        return $this;
    }

    public function getEquipement(): ?Equipement
    {
        return $this->equipement;
    }

    public function setEquipement(Equipement $equipement): self
    {
        $this->equipement = $equipement;

        return $this;
    }

    public function getQuantite(): ?int
    {
        return $this->quantite;
    }

    public function setQuantite(int $quantite): self
    {
        $this->quantite = $quantite;

        return $this;
    }

    public function getDateCreation(): ?\DateTimeInterface
    {
        return $this->dateCreation;
    }

    public function setDateCreation(\DateTimeInterface $dateCreation): self
    {
        $this->dateCreation = $dateCreation;

        return $this;
    }
}